miner/Miner.hs

```diff
@@ -263,7 +285,72 @@ scheme :: Env -> (TargetBytes -> HeaderBytes -> RIO Env HeaderBytes)
 scheme env = case envCmd env of
     CPU e _ -> cpu e
     GPU e _ -> gpu e
-    Keys -> error "Impossible: You shouldn't reach this case."
+    _ -> error "Impossible: You shouldn't reach this case."
+
+
+{- this will be ugly -}
+genLocal
+    :: ChainwebVersion
+    -> ChainId
+    -> P.Command Text
+    -> ClientM (P.CommandResult Hash)
+genLocal version cid =
+  case someChainwebVersionVal version of
+    SomeChainwebVersionT (_ :: Proxy cv) ->
+      case someChainIdVal cid of
+        SomeChainIdT (_ :: Proxy ccid) ->
+          client (Proxy :: Proxy (PactLocalApi cv ccid))
+
+getBalances :: BaseUrl -> Miner -> IO ()
+getBalances url (Miner (MinerId mi) _) = do
```

Comment:
If you're just after an identifier here, there's no reason to ask for the full `Miner` type. You really just need the account name. All balances and account names are public data, so this is fine.